app/helpers/close_contact_query_helper.rb

```diff
@@ -5,4 +5,102 @@ module CloseContactQueryHelper
+    end
+
+    if sort_order.present? && !%w[first_name last_name primary_telephone email last_date_of_exposure assigned_user contact_attempts enrolled_id
+                                  notes].include?(sort_order)
+      error_message = "Unable to sort by specified column in request: '#{sort_order}'"
+      raise StandardError, error_message
+    end
+
+    # Either both the sort order and direction are provided, or neither is provided
+    if (sort_direction.blank? && sort_order.present?) || (sort_direction.present? && sort_order.blank?)
+      error_message = 'Must have both a sort column and direction specified or neither specified. Requested column to sort: '\
+                      "'#{sort_order}'', with specified direction: '#{sort_direction}'"
+      raise StandardError, error_message
+    end
+
+    raise InvalidQueryError.new(:entries, entries) if entries.negative?
+    raise InvalidQueryError.new(:page, page) if page.negative?
+
+    {
+      patient_id: patient_id,
+      search_text: search_text,
+      sort_order: sort_order,
+      sort_direction: sort_direction,
+      entries: entries,
+      page: page
+    }
+  end
+
+  # Queries close_contact by ID or type based on search text.
+  def search(close_contact, search)
+    return close_contact if search.blank?
+
+    close_contact.where('lower(first_name) like ?', "#{search&.downcase}%").or(
+      close_contact.where('lower(last_name) like ?', "#{search&.downcase}%").or(
+        close_contact.where('primary_telephone like ?', "%#{search}%").or(
+          close_contact.where('lower(email) like ?', "#{search&.downcase}%").or(
+            close_contact.where('assigned_user like ?', "#{search&.downcase}%").or(
+              close_contact.where('contact_attempts like ?', "#{search&.downcase}%")
+            )
+          )
+        )
+      )
+    )
+  end
+
+  # Sorts close_contacts based on given column and direction.
+  def sort(close_contacts, order, direction)
+    # Order by created_at date by default
+    return close_contacts.order(created_at: 'desc') if order.blank? || direction.blank?
+
+    # Satisfy brakeman with additional sanitation logic
+    dir = direction == 'asc' ? 'asc' : 'desc'
+
+    case order
+    when 'first_name'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN first_name IS NULL THEN 1 ELSE 0 END, first_name ' + dir))
+    when 'last_name'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN last_name IS NULL THEN 1 ELSE 0 END, last_name ' + dir))
+    when 'primary_telephone'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN primary_telephone IS NULL THEN 1 ELSE 0 END, primary_telephone ' + dir))
+    when 'email'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN email IS NULL THEN 1 ELSE 0 END, email ' + dir))
+    when 'enrolled_id'
+      close_contacts = close_contacts.order(Arel.sql('CASE WHEN enrolled_id IS NULL THEN 1 ELSE 0 END, enrolled_id ' + dir))
```

Comment:
Since the corresponding column in the table in the UI is actually just a "Yes" or "No", we'd want this to sort based on whether the close contact is enrolled rather than the ID itself.